### PR TITLE
Various enhancements

### PR DESCRIPTION
**Pull request by starfys**
* Cleaned up code
* Added dank memes
* upgraded dependencies 
* Made zubotsu compatible with new serenity version.
* Removed unnecessary clones
* Modularized the emojifier

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -180,49 +130,68 @@ impl Framework for ZubotsuFramework {
...
-                            }
-                            None => Ok(()),
-                        };
+                    for emoji in emoji::emojify(message_text) {
+                        let _ = message.react(&context,  emoji);
                     }
                 }
             }
-
+            // @@@@&%####(((((((((((((((//////////////////////////////(((##%%&&&&@@@@@@@@@@@@@
```
Comment (Riuchando) [inline src/main.rs]: oh no

File: src/emoji.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    emoji_map.insert('6', vec!["6⃣"]);
+    emoji_map.insert('5', vec!["5⃣"]);
+    emoji_map.insert('4', vec!["4⃣"]);
+    emoji_map.insert('3', vec!["3⃣"]);
+    emoji_map.insert('2', vec!["2⃣"]);
+    emoji_map.insert('1', vec!["1⃣"]);
+    emoji_map.insert('0', vec!["0⃣"]);
+    
+
+    // TODO: may make it easy to just make this a hashset
```
Comment (Riuchando) [inline src/emoji.rs]: would we make this a hashset and then remove the item from the set after it is used?

File: src/main.rs [hunk 3]
```diff
@@ -14,7 +14,11 @@
 // You should have received a copy of the GNU General Public License
 // along with Zubotsu.  If not, see <https://www.gnu.org/licenses/>.
 
+mod data;
+mod emoji;
+
 use chrono::prelude::*;
+use log::{debug, info};
```
Comment (Riuchando) [inline src/main.rs]: if we have logging on the server now, is it better to start handling the various error that we have, or maybe put that in a separate pr?

File: src/emoji.rs (hunk 2 above)
Comment (sremedios) [reply src/emoji.rs]: agreed with @Riuchando

File: src/main.rs [hunk 4]
```diff
@@ -222,7 +222,45 @@ impl Framework for ZubotsuFramework {
...
+            // ****/(((#%%%%%%%%%%%%%%%%##((###((########%%%%%%#####%%%#######%%%%%%%%%%&&%%#(
+            // //////((%%%%%%#####%%%%%%#########%%%%%%%%%%####(((###########%%%%%%%%%&&&%%#(
+            // #########%%&%%#%%####%%%%%%%%%%####%%%##%######(((((##########%%%%%%%%&&&&%%#((
+            // ###%%%%%%%%&%%%#####%%#%%#################(#((((((((##########%%%%%%&&&&&&%%#((
+            // **////////(#&&%%%%%###%%%#################(((((((((##########%%%%%&&&&&&&%##(((
+            // ....,,,,,,,*(#%%%%%%####%%%%%#############(((((((((########%%%%%&&&&&@@&%%##(((
+            // ,,.......,,,,*#%%%%%#####%%%%%%%%%%######((((((((#########%%%%%&&&&@@@&&%#(((((
+            // *,,,......,,,,/%&&&%########%#%%%%%#####(((((((((((#%%%%%%%%%%&&&&@@@&&%##(((((
+            // /***,,,,,,,,,**(%&&%%###########%%%####((((((((((###%%%%%%%%&&&&&&&&&%%##((((((
             if message_text == "!megadownbishoy" {
```
Comment (sremedios) [inline src/main.rs]: Should this be somewhat occluded for privacy?

File: src/main.rs (hunk 3 above)
Comment (starfys) [reply src/main.rs]: separate PR IMO

File: src/main.rs (hunk 4 above)
Comment (starfys) [reply src/main.rs]: d o n e
